File: work/abc080_d.cpp
#include<iostream>
#include<vector>
using namespace std;

int main(){
  int N,C;
  cin>>N>>C;
  int T = 0;
  vector<int> s(N);
  vector<int> t(N);
  vector<int> c(N);
  for(int i=0;i<N;++i){
    cin>>s[i]>>t[i]>>c[i];
    if(t[i]>T){
      T = t[i];
    }
  }

  vector<int> rm(T,0);
  for(int ci=0;ci<C;++ci){
    vector<int> tt(T,0);
    for(int i=0;i<N;++i){
      if(c[i]==ci+1){
        tt[s[i]-1] += 1;
        tt[t[i]-1] -= 1;
      }
    }
    for(int ti=1;ti<T;++ti){
      if(tt[ti]==1 && tt[ti-1]==0){
        tt[ti] = 0;
        tt[ti-1] = 1;
      }
    }
    for(int ti=0;ti<T-1;++ti){
      tt[ti+1] = tt[ti+1]+tt[ti];
    }
    for(int ti=0;ti<T;++ti){
      if(tt[ti]>0){
        rm[ti] += 1;
      }
    }
  }
  int ret = 0;
  for(int i=0;i<T;++i){
    if(ret<rm[i]){
      ret = rm[i];
    }
  }
  cout<<ret<<endl;
}

File: work/abc080_d.py
# -*- coding: utf-8 -*-
n,c = map(int, input().split())
tv = []
T = 0
for _ in range(n):
    line = list(map(int, input().split()))
    T = max(T, line[1])
    tv.append(line)

rm = [0 for _ in range(2*T)]

for ci in range(c):
    tt = [0 for _ in range(2*T)]
    for i in range(n):
        if tv[i][2]==ci+1:
            tt[2*tv[i][0]-2] += 1
            tt[2*tv[i][1]-1] -= 1
    for t in range(2*T-1):
        tt[t+1] = tt[t+1]+tt[t]
    for t in range(2*T):
        if tt[t]>0:
            rm[t] += 1
    # print(tt)
# print(rm)
ret = 0
for s in rm:
    ret = max(ret, s)
print(ret)
